### DISCUSSION TITLE
Query for Camomile or Pd environment

### DISCUSSION THREAD

**Discussion by GModal** [pierreguillot/Camomile#249] [Q&A] [answered]
Is there a straight-forward way to know if a patch is running on Camomile or in the Pure Data prog/interface? I'd like a single patch to work on both, and some functions (openpanel, for instance) work differently.

I'm guessing one way would be by checking if any _param_ data is being received.

**Comment by pierreguillot** [comment, marked as answer]
There is no way to be sure that the patch is loaded via Camomile. Nevertheless, you can use Camomile standard messages to assume that the patch is loaded as a plugin (the *param* message, if the plugin has a least one parameter or better, the *bus* message that is sent when starting the DSP if the plugin has at least one bus). Or another solution would be to check the path of the patch using *pdcontrol* (for example on macOS, the path should contain */Contents/Resources/*).

**Reply by GModal** [reply]
Thanks -- that's a good tip about the _bus_ message. I'll go with that, it seems simpler.
